### DISCUSSION TITLE
Can't get superset to connect to mysql database

### DISCUSSION THREAD

**Discussion by saqeria** [apache/superset#11895] [General]
Dear,

I've tried too many methods, different docker repository's and versions but I can't get superset to work with MYSQL. I made a MYSQL container and checked if all the ports and database info is correct. When I try to add the SQLAlchemy URI and test the connection it says the "ERROR: Connection failed, please check your connection settings" error. I've read through the repository setup/confiquration. But unfortunately nothing works. I'm trying to set up superset in synology docker. If anyone could please help me out it would be very much appreciated, currently I've been searching and trying different things for 4 hours now.. I know this isn't an issue and it's probably an issue on my side, but I'm really trying my best to set it up. 

Some info:
- I didn't load a custom superset_config.py as I've read that it isn't really necessary.
- I've changed the MYSQL database port to 3309 because port 3306 was already in use by another container.
- The SQLAlchemy URI (removed personal info): mysql://User:Password@localhost:3309/database

Thanks in advance for your effort.
Sorry for being stupid, I really tried to learn and research as much as possible before opening this "issue"
Help is really much appreciated 😢

**Comment by JessicaJiang98** [comment]
I am having the same problem right now... Wondering if you have solved it.

**Comment by srinify** [comment]
Hmm are all these services (Superset & MySQL in particular) running within Docker? I would try replacing `localhost` with `host.docker.internal`. It's a Docker trick!

For example, here's my local connection details for CrateDB where I use this trick ^

![image](https://user-images.githubusercontent.com/801507/112554532-dc937d00-8d9c-11eb-856e-a52dcb57b54f.png)

**Reply by Coding-Deng** [reply]
666

**Comment by duchuynh** [comment]
> Hmm are all these services (Superset & MySQL in particular) running within Docker? I would try replacing `localhost` with `host.docker.internal`. It's a Docker trick!
> 
> For example, here's my local connection details for CrateDB where I use this trick ^
> 
> ![image](https://user-images.githubusercontent.com/801507/112554532-dc937d00-8d9c-11eb-856e-a52dcb57b54f.png)

Hi,

Where would you get the SQL Alchemy URL for this problem?
